tests/test_session_base.py

```
@@ -738,6 +754,101 @@ def test_run_no_container(self) -> None:
+        """Existing advisory behavior should remain the default."""
+        pattern = SecurityPattern(
+            pattern="dangerous", description="Dangerous code", severity=SecurityIssueSeverity.HIGH
+        )
+        policy = SecurityPolicy(patterns=[pattern], severity_threshold=SecurityIssueSeverity.MEDIUM)
+        config = SessionConfig(security_policy=policy)
+        session = MockBaseSession(config)
+        session.container = Mock()
+        session.is_open = True
+
+        mock_file = Mock()
+        mock_file.name = "/tmp/code.py"
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/code.py"

(S108)